src/syngen/ml/convertor/convertor.py

```diff
@@ -24,29 +25,40 @@ def _preprocess_df(schema: Dict, df: pd.DataFrame) -> pd.DataFrame:
         Preprocess data frame, update data types of columns
         """
         if not df.empty:
-            for column, data_type in schema.get("fields", {}).items():
-                if data_type in ["binary", "date"]:
-                    df[column] = df[column].astype("string")
-                elif data_type == "int":
-                    if any(df[column].isnull()):
-                        df[column] = df[column].astype("float64")
-                    else:
-                        df[column] = df[column].astype("int64")
-                elif data_type == "string":
-                    df[column] = df[column].astype("string")
+            if schema["format"] != "CSV":
```

Comment:
It looks like the logic of this function related to CSVConvertor should be defined separately in the method '_preprocess_df' in CSVConvetor in order to avoid keeping all this logic in one place.

Reply:
I thought it is little strange to define an AVRO-related logic in the base class. Maybe it would be better to whether live it as blank placeholder in the base class and declare in each child or make it universal and suitable for all children?